src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -508,8 +508,8 @@ event::ptr primitive_inst::realloc_if_needed() {
     }
 
     // Clear out memory if if was previously reused, but now primitive can't be optimized
-    if (_node->is_type<gather>() && !can_be_optimized() && _outputs[0]
-            && dep_memory_ptr(0) && _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
+    if ((_node->is_type<gather>() || _node->is_type<permute>()) && !can_be_optimized() && _outputs[0]
+        && dep_memory_ptr(0) && _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
         _outputs[0] = nullptr;
```

Comment:
Also, lets let it return if it is can_be_optimized(), not allocating memory..

Reply:
Updated to return an empty event when can_be_optimized() is true